include/cell.tcc

```diff
@@ -844,3 +844,17 @@ template <unsigned Tdim>
 inline unsigned mpm::Cell<Tdim>::previous_mpirank() const {
   return this->previous_mpirank_;
 }
+
+//! Map cell volume to nodes
+template <unsigned Tdim>
+void mpm::Cell<Tdim>::map_cell_volume_to_nodes(unsigned phase) {
+  if (this->status()) {
```

Comment:
We only do this for cells with particles as otherwise, the `nodes_->update_volume()` will be called as many numbers as the MPI thread. Therefore, without checking the `status()`, we won't have the free_surface detection working in parallel.